scripts/ci/get_modified_tests.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+
+# A script to generate a list of tests that have changed or added and create an
+# arguemnts file for sanitycheck to allow running those tests with --all
+
```

Comment:
So what is the logic on this? it seems to me that only if there is a change in (testcase|sample).yaml it is caught.

I'd start with "any changes in samples/ or tests/ have to be looked at" -- now . I'd suggest if the path matches (so a leaf file):

"(samples|tests)/.+/.+/[^/]+"

then the path has to be added to a set (not a list), as multiple matches (let's say in the Makefile or a source file etc) wouldn't have to result in multiple additions.